src/server.cpp

```diff
@@ -1,17 +1,79 @@
 #include "crow.h"
+#include <string>
+#include <vector>
+#include <boost/thread.hpp>
+
+std::vector<std::string> locks;
+boost::mutex searchLock;
+
+bool getLock(std::string lockName) 
+{
+  searchLock.lock();
+  for (long unsigned int i = 0; i < locks.size(); i++) 
+  {
+    if (lockName == locks[i]) 
+    {
+      return false;
+    }
+  }
+
+  locks.push_back(lockName);
+  return true;
+}
+
+bool releaseLock(std::string lockName)
+{
+  searchLock.lock();
+  for (long unsigned int i = 0; i < locks.size(); i++) 
+  {
+    if (lockName == locks[i]) 
+    {
+      locks.erase(locks.begin() + i);
+      searchLock.unlock();
+      return true;
+    }
+  }
+  searchLock.unlock();
+  return false;
+}
 
 int main()
 {
+    
     crow::SimpleApp app;
+    CROW_ROUTE(app, "/")([]()
+    {
+        return "Hello LockWorld";
+    });
 
-    CROW_ROUTE(app, "/")([](){
-        return "Hello, LyingDutch";
+    CROW_ROUTE(app, "/getLock/<string>") ( [](std::string lockName) 
+    {   
+        if (!getLock(lockName)) 
+        {
+          searchLock.unlock();
```

Comment:
Why doesn't `getLock` take care of releasing the mutex, while `releaseLock` does? 
I would say that `releaseLock` has the right behaviour (i.e. not leaving the responsibility to close a lock to the caller).